runelite-client/src/main/java/net/runelite/client/plugins/cannon/CannonPlugin.java

```diff
@@ -198,6 +206,26 @@ public void onItemContainerChanged(ItemContainerChanged event)
 			}
 		}
 
+		// After just placing a cannon and loading balls into it.
+		// this allows reclaimed cannons with balls still loaded
+		// to have cballsLeft initialized correctly.
+		if (inventorySnapshot != null)
+		{
+			Multiset<Integer> diff = generateInventoryDiff();
+			int snapshotCannonballs = inventorySnapshot.count(ItemID.CANNONBALL);
+			int diffCannonballs = diff.count(ItemID.CANNONBALL);
+			int snapshotGraniteCannonballs = inventorySnapshot.count(ItemID.GRANITE_CANNONBALL);
+			int diffGraniteCannonballs = diff.count(ItemID.GRANITE_CANNONBALL);
+
+			if ((diffCannonballs > 0 && snapshotCannonballs > diffCannonballs)
+					|| (diffGraniteCannonballs > 0 && snapshotGraniteCannonballs > diffGraniteCannonballs))
```

Comment:
good shout, i think this can actually just become `diff.contains(ItemID.CANNONBALL) && inventory.contains(ItemID.CANNONBALL)` then, since all that matters is the diff and current inventory both contain cannonballs in them